More expressive titles in the change log (Lombiq Technologies: OCORE-199)

Piedone opened the pull request:
Following up from https://github.com/OrchardCMS/OrchardCore/pull/16755#discussion_r1765982886.

MikeAlhayek commented:
I still think we should not do this. Our current format groups all the changes that are related to the same feature under one bullet point.

IMO, I think we should still continue to follow the following format:

```
# Release Version

## Breaking Changes

### Feature Name
- Change 1
- Change 2
- Change 3

## Change Logs

### Feature Name
- Change 1
- Change 2
- Change 3
```

Piedone commented:
Fine by me, how about now?

MikeAlhayek commented on src/docs/releases/2.1.0.md:
```diff
@@ -6,6 +6,8 @@ Release date: Not yet released
 
 ### External Authentication Feature
 
+#### External Authentication Logic Factored out from the Users Feature
```
Maybe move the `### External Authentication Feature` to a new section **Added Features**

```suggestion
#### External authentication logic has been separated from the Users feature.
```

Piedone replied:
This is a title, so should be capitalized like titles, and it's not a sentence thus shouldn't end with punctuation.

MikeAlhayek replied:
fine. the suggestion was updated

MikeAlhayek commented on src/docs/releases/2.1.0.md:
```diff
@@ -32,11 +34,13 @@ Also, note the following updates in `ExternalLoginSettings`:
 
 ### User Registration Feature
 
+#### User Registration Feature Not Needed for External Authentication
```
```suggestion
#### **User Registration** Feature No Longer Required for External Authentication
```

Piedone replied:
Formatting with `**` won't show a difference in how the document site is rendered. Neither should we really try to change the formatting of titles one-off.

MikeAlhayek replied:
right because it's already highlighted. updated.

MikeAlhayek commented on src/docs/releases/2.1.0.md:
```diff
@@ -52,16 +56,22 @@ Previously, the `UsersCanRegister` property controlled which types of registrati
     Please review all your recipes and replace occurrences of `AssignRoles` with `AssignRoleToUsers`, and `AssignRole_{RoleName}` with `AssignRoleToUsers_{RoleName}`.
 
-### New **Azure Communication Services SMS** Feature
+### Azure Communication Services SMS Feature
+
+#### New Azure Communication Services SMS Feature
```
Also move this section to the bottom of the file after line 77. This way we can highlight new features away from change logs.

```suggestion

## Added Features

### Azure Communication Services SMS Feature

#### New **Azure Communication Services SMS** Feature
```

Piedone replied:
If we have an "Added Features" section, then having at least two levels for each feature is redundant, so I added it without it. See my note about formatting titles above.

MikeAlhayek commented:
@Piedone I would also either move the addition `External Authentication` feature to `New Features` section, or add a reference to the new `External Authentication` feature in the `New Features` section.

Piedone commented:
I added it at the bottom. However, now the original section above doesn't make sense. So, I moved that under the Users feature, which is the one that actually changed.

MikeAlhayek commented on src/docs/releases/2.1.0.md:
```diff
@@ -52,17 +56,27 @@ Previously, the `UsersCanRegister` property controlled which types of registrati
+
+### External Authentication Feature
+
+The new **External Authentication** feature contains common functionality used by multiple external (i.e. non-local) authentication providers (like Microsoft or Google login). See [above](#external-authentication-logic-has-been-separated-from-the-users-feature) for details.
```
```suggestion
The new **External Authentication** feature contains common functionality used by multiple external (i.e. non-local) authentication providers (like Microsoft or Google login). See [above](#external-authentication-logic-has-been-separated-from-the-users-feature) for details.

```
